packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.scss

```diff
@@ -6,220 +6,239 @@
-      border-color: $ms-color-themeDarkAlt;
-    }
-  }
-
-  &:active {
-    &::before {
-      border-color: $ms-color-neutralSecondaryAlt;
-    }
-
-    .textLabel {
-      color: $ms-color-black;
-    }
-  }
-
-  //== State: A checkbox is checked
-  //
-  &.labelIsChecked {
-    &::before {
-      border: 10px solid $ms-color-themePrimary;
-      background-color: $ms-color-themePrimary;
-
-      @media screen and (-ms-high-contrast: active), (-ms-high-contrast: black-on-white) {
-        display: none;
-      }
-    }
-
-    &::after {
-      display: block;
-
-      @media screen and (-ms-high-contrast: active), (-ms-high-contrast: black-on-white) {
-        height: $ms-checkbox-highContrast-iconBoxSize;
-        width: $ms-checkbox-highContrast-iconBoxSize;
-        line-height: $ms-checkbox-highContrast-iconBoxSize;
-      }
-
-      @include highContrast-border($ms-color-white, $ms-color-black);
+    overflow: hidden;
+    position: relative;
+    display: flex;
+    align-items: center;
+
+    > .input {
+        margin: 0px;
+        height: 0px;
+        width: 0px;
+        visibility: hidden;
+        opacity: 0;
+    }
+
+    > .label {
+        display: flex;
+        align-items: center;
+        margin: 0 -4px;
+        cursor: pointer;
+
+        > * {
+            margin: 0 4px;
+        }
+
+        > .box {
+            display: flex;
+            align-items: center;
+            justify-content: center;
+            flex: 0 0 $ms-checkbox-label-size;
+            height: $ms-checkbox-label-size;
+
+            > .checkbox {
+                display: flex;
+                align-items: center;
+                justify-content: center;
+                height: $ms-checkbox-label-size;
+                width: $ms-checkbox-label-size;
+                border: 1px solid black;
+                box-sizing: border-box;
+                transition-property: background, border, border-color;
+                transition-duration: $ms-checkbox-transition-duration;
+                transition-timing-function: $ms-checkbox-transition-timing;
+                /* incase the icon is bigger than the box */
+                overflow: hidden;
+
+                @include ms-borderColor-neutralTertiary;
+                @media screen and (-ms-high-contrast: active) {
+                  border-color: $ms-color-contrastWhiteDisabled;
+                }
+
+                @media screen and (-ms-high-contrast: black-on-white) {
+                  border-color: $ms-color-contrastBlackDisabled;
+                }
+
+                > .checkmark {
+                    opacity: 0;
+                    flex: 0 0 auto;
+                    @include ms-fontColor-white;
+                }
+            }
+        }
+
+        > .text {
+            flex: 1 0 auto;
+        }
     }
 
     &:hover,
+    &:active,
     &:focus {
-      &::before {
-        border-color: $ms-color-themeDarkAlt;
-      }
+        > .label {
+            > .box > .checkbox {
```

Comment:
Still not sure why we have all this nesting. You shouldn't need it, with css modules being local and unique.